pymc3/quadratic_approximation.py

```diff
@@ -0,0 +1,68 @@
+#   Copyright 2020 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Functions for Quadratic Approximation."""
+
+import numpy as np
+import scipy
+import arviz as az
+
+__all__ = [
+    "quadratic_approximation"
+]
+
+from pymc3.tuning import find_MAP, find_hessian
+
+
+def quadratic_approximation(vars, n_chains=2, n_samples=10_000):
+    """ Finds the quadratic approximation to the posterior, also known as the laplace approximation.
+
+    NOTE: The quadratic approximation only works well for unimodal and roughly symmetrical posteriors of continuous variables.
+    The usual MCMC convergence and mixing statistics (e.g. R-hat, ESS) will NOT tell you anything about how well this approximation fits your actual (unknown) posterior, indeed they'll always be extremely nice since all "chains" are sampling from exactly the same distribution, the posterior quadratic approximation.
+    Use at your own risk.
+
+    See Chapter 4 of "Bayesian Data Analysis" 3rd edition for background.
+
+    Returns an arviz.InferenceData object for compatibility by sampling from the approximated quadratic posterior. Note these are NOT MCMC samples, so the notion of chains is meaningless, and is only included for downstream compatibility with Arviz.
+
+    Also returns the exact posterior approximation as a scipy.stats.multivariate_normal distribution.
+
+    Parameters
+    ----------
+    vars: list
+        List of variables to approximate the posterior for.
+    n_chains: int
```

Comment:
I don't think we need this kwarg and should fix it to 1.

Reply:
Unfortunately, Arviz will complain if there aren't at least 2 chains.

Reply:
can you share an example? I think you'll get trouble for multivariate distributions, where arviz will interpret the leading dimension as the chains, but I would solve that as follows:

```python
data = {'a': np.ones(10), 'b': np.ones((100, 3))}

az.convert_to_inference_data(data) # warning, misinterprets `b` as having 100 chains

az.convert_to_inference_data({k: v[np.newaxis, ...] for k, v in data.items()}) # Good, explicitly sets number of chains to 1.
```

But maybe I'm misunderstanding the issue!

I'm against including chains since it gives the impression that you might want to run diagnostics on the quality of the returned samples.

Reply:
![image](https://user-images.githubusercontent.com/206013/125208435-918d6f00-e292-11eb-9ae5-d42b6c0c948b.png)

I guess that's not too bad. All the plots seem to work. I just saw that warning and figured nothing would work. I'll remove the chains. I fully agree. The less impression we can give that running diagnostics the better. A little warning and a NaN might actually be good :)